src/main/kotlin/io/kotest/plugin/intellij/tests/KotestTestGenerator.kt

```diff
@@ -40,18 +45,25 @@ class KotestTestGenerator : JavaTestGenerator() {
       // the framework name and delegate to the JavaTestGenerator one. But this only works while JavaTestGenerator
       // is public, and has a zero arg constructor, so I would like to find a better long term solution:
       // https://intellij-support.jetbrains.com/hc/en-us/community/posts/15040678418450-How-to-choose-when-to-apply-TestGenerator
-      if (d.selectedTestFrameworkDescriptor.name != Constants.FRAMEWORK_NAME)
-         return super.generateTest(project, d)
-      return PostprocessReformattingAspect.getInstance(project).postponeFormattingInside(Computable {
-         ApplicationManager.getApplication().runWriteAction(Computable<PsiElement?> {
-            val file = generateTestFile(project, d)
-            if (file != null) {
-               // without this the file is created but the caret stays in the original file
-               CodeInsightUtil.positionCursor(project, file, file)
-            }
-            file
+      return if (d.selectedTestFrameworkDescriptor.name != Constants.FRAMEWORK_NAME) {
+         val testGenerator = if (KotlinPluginModeProvider.isK2Mode()) {
+            kotlinTestGenerator
+         } else {
+            javaTestGenerator
```

Comment:
won't the generator be the same either way, given that
 private val kotlinTestGenerator = KotlinTestGenerator()
   private val javaTestGenerator = KotlinTestGenerator()

Reply:
oh! good catch, indeed!